core/interop/src/try_into_js_result_impls.rs

```diff
@@ -0,0 +1,55 @@
+//! Declare implementations of [`TryIntoJsResult`] trait for various types.
+//! We cannot rely on a generic implementation based on `TryIntoJs` due
+//! to a limitation of the Rust type system which prevents generalization
+//! of traits based on an upstream crate.
```

Comment:
If this is the reason why we can't rely on `TryIntoJs`, maybe this should be moved to the main crate?

Reply:
Sure, I'll make this, but probably just reimplement `TryInto<JsResult<>>` if possible. That will be simpler. I was going to seal this trait but no need, and being able to extend it is useful.